src/lib/routes/admin-api/telemetry.ts

```diff
@@ -0,0 +1,67 @@
+import { Response } from 'express';
+import { OpenApiService } from 'lib/services';
+import { IAuthRequest } from '../unleash-types';
+import { IUnleashConfig } from '../../types/option';
+import Controller from '../controller';
+import { NONE } from '../../types/permissions';
+import { IUnleashServices } from 'lib/types';
+import { createResponseSchema } from '../../openapi/util/create-response-schema';
+import {
+    telemetrySettingsSchema,
+    TelemetrySettingsSchema,
+} from '../../openapi/spec/telemetry-settings-schema';
+
+class TelemetryController extends Controller {
+    config: IUnleashConfig;
+
+    openApiService: OpenApiService;
+
+    constructor(
+        config: IUnleashConfig,
+        { openApiService }: Pick<IUnleashServices, 'openApiService'>,
+    ) {
+        super(config);
+        this.config = config;
+        this.openApiService = openApiService;
+
+        this.route({
+            method: 'get',
+            path: '/settings',
+            handler: this.getTelemetrySettings,
+            permission: NONE,
+            middleware: [
+                openApiService.validPath({
+                    tags: ['Telemetry'],
+                    summary:
+                        'Get settings for telemetry information collection',
```

Comment:
This could probably be a bit shorter to make it easier to find. We provide more info in the description anyway.

```suggestion
                        'Get telemetry settings',
```